### PR TITLE
ci(propagate-content): use boolean inputs

### PR DESCRIPTION
**Pull request by educhastenier**
as it is easier to handle boolean inputs through check boxes

### CODE REVIEW

File: .github/workflows/propagate-doc-upwards.yml [hunk 1]
```diff
@@ -5,13 +5,13 @@ on:
...
+        type: boolean
+        description: 'Dry run: if checked, run the merge locally but do not push to remote'
+        default: false
       slack-notifications:
-        description: 'If true, send slack notifications when errors occur'
-        required: false
-        default: 'false'
+        type: boolean
+        description: 'If checked, send slack notifications when errors occur'
+        default: true
```
Comment (educhastenier) [inline .github/workflows/propagate-doc-upwards.yml:14]: I think it is better to change this default value to 'true'.
BTW, how did it work in the cron, as we can see that notifications are correctly sent by default?

Comment (benjaminParisel) [reply .github/workflows/propagate-doc-upwards.yml:14]: The condition for the Slack notification in the workflow.
`if: failure() && (github.event_name == 'schedule' || (github.event_name == 'workflow_dispatch' && github.event.inputs.slack-notifications == 'true') )`

So when the cron starts, the inputs.slack-notifications are ignored.

Comment (tbouffard) [reply .github/workflows/propagate-doc-upwards.yml:14]: About the default value, is set to false en purpose. The input has been created mainly to troubleshoot slack issue, but not to send a message when the workflow manually.

When someone is triggering the workflow manually, it is supposed to monitor the run by themself. There is no need to pollute other Engineers with a slack notification.

Comment (tbouffard) [inline .github/workflows/propagate-doc-upwards.yml:14]:
```suggestion
        default: false
```

File: .github/workflows/propagate-doc-upwards.yml [hunk 2]
```diff
@@ -55,7 +55,7 @@ jobs:
           ./.././bonita-documentation-site/scripts/propagate_doc_upwards.bash
 
       - name: Send message to Slack channel
-        if: failure() && (github.event_name == 'schedule' || (github.event_name == 'workflow_dispatch' && github.event.inputs.slack-notifications == 'true') )
```
Comment (tbouffard) [inline .github/workflows/propagate-doc-upwards.yml:58]: **question:** did you test the change by running the action? AFAIK, whatever the type is defined in the input, the actual type at runtime is always a string, so this change may change the behaviour (the input is always set, so the condition would always be true)

Comment (educhastenier) [reply .github/workflows/propagate-doc-upwards.yml:58]: My bad, reverting it...

**Review by tbouffard — APPROVED** [review]
✔️ Tested with https://github.com/bonitasoft/bonita-documentation-site/actions/runs/8112514456 using the branch of this PR. Dry-run and no slack notifications.

image

**Pull request merged**